app/controllers/schemes_controller.rb

```diff
@@ -162,17 +162,24 @@ def scheme_params
                                                      :secondary_client_group,
                                                      :support_type,
                                                      :support_services_provider,
+                                                     :support_services_provider_before_type_cast,
                                                      :intended_stay)
 
-    same_org_providing_support = required_params[:support_services_provider] == "The same organisation that owns the housing stock"
+    same_org_providing_support = required_params[:support_services_provider_before_type_cast] == "0"
```

Comment:
What about:

```ruby
 required_params = params.require(:scheme).permit(:service_name,
                                                     :sensitive,
                                                     :owning_organisation_id,
                                                     :managing_organisation_id,
                                                     :scheme_type,
                                                     :registered_under_care_act,
                                                     :id,
                                                     :has_other_client_group,
                                                     :primary_client_group,
                                                     :secondary_client_group,
                                                     :support_type,
                                                     :support_services_provider_before_type_cast,
                                                     :intended_stay)
 params = required_params.except("support_services_before_type_cast")
 if required_params["support_services_provider_before_type_cast"]
    params.merge(Scheme::SUPPORT_SERVICES_PROVIDER.invert[required_params["support_services_provider_before_type_cast"]]) 
end
params
```

Then you can use `scheme_params` as before everywhere else in the controller without needing to change anything else?

Reply:
I will most certainly try this today